protobuf-converter/src/main/java/io/confluent/connect/protobuf/ProtobufData.java

```diff
@@ -0,0 +1,932 @@
+    try {
+      if (isProtobufTimestamp(schema)) {
+        DynamicMessage message = (DynamicMessage) value;
+
+        long seconds = 0L;
+        int nanos = 0;
+        for (Map.Entry<FieldDescriptor, Object> entry : message.getAllFields().entrySet()) {
+          if (entry.getKey().getName().equals("seconds")) {
+            seconds = ((Number) entry.getValue()).longValue();
+          } else if (entry.getKey().getName().equals("nanos")) {
+            nanos = ((Number) entry.getValue()).intValue();
+          }
+        }
+        com.google.protobuf.Timestamp timestamp = com.google.protobuf.Timestamp.newBuilder()
+            .setSeconds(seconds)
+            .setNanos(nanos)
+            .build();
+        return Timestamp.toLogical(schema, Timestamps.toMillis(timestamp));
+      }
+
+      Object converted = null;
+      switch (schema.type()) {
+        case INT8:
+        case INT16:
+        case INT32:
+          if (value instanceof Number) {
+            converted = ((Number) value).intValue(); // Validate type
+          } else if (value instanceof Enum) {
+            converted = ((Enum) value).ordinal();
+          } else if (value instanceof EnumValueDescriptor) {
+            converted = ((EnumValueDescriptor) value).getNumber();
+          }
+          break;
+        case INT64:
+          long longValue;
+          if (value instanceof Long) {
+            longValue = (Long) value;
+          } else {
+            longValue = Integer.toUnsignedLong(((Number) value).intValue());
+          }
+          converted = longValue;
+          break;
+        case FLOAT32:
+          float floatValue = ((Number) value).floatValue(); // Validate type
+          converted = value;
+          break;
+        case FLOAT64:
+          double doubleValue = ((Number) value).doubleValue(); // Validate type
+          converted = value;
+          break;
+        case BOOLEAN:
+          Boolean boolValue = (Boolean) value; // Validate type
+          converted = value;
+          break;
+        case STRING:
+          if (value instanceof String) {
+            converted = value;
+          } else if (value instanceof CharSequence) {
+            converted = value.toString();
+          } else {
+            throw new DataException("Invalid class for string type, expecting String or "
+                + "CharSequence but found "
+                + value.getClass());
+          }
+          break;
+        case BYTES:
+          if (value instanceof byte[]) {
+            converted = ByteBuffer.wrap((byte[]) value);
+          } else if (value instanceof ByteBuffer) {
+            converted = value;
+          } else if (value instanceof ByteString) {
+            converted = ((ByteString) value).asReadOnlyByteBuffer();
+          } else {
+            throw new DataException("Invalid class for bytes type, expecting byte[] or ByteBuffer "
+                + "but found "
+                + value.getClass());
```

Comment:
Can value be null here ?

Nit: ByteString is also accepted.

Reply:
Ack